code-challenges.test.js

```diff
@@ -23,30 +39,86 @@ const secretCodeWord3 = "Eccentric"
 
 // b) Create the function that makes the test pass.
 
+const codedString = (str) => {
+    let code = {"a":4, "e":3, "i":1, "o":0}
+    return str.split("").map((value) =>{
+        return(object.keys(code).includes(value.toLowercase())) ? code[value.toLowercase()] : value
+    }).join("")
+}
+
+// pseudo code: 
+// input: function that takes in a string
+// output/return: a string with a coded message. The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.
+// I created a function that took in a string and returned an array of strings with the split "" and used the .map to iterate through every element in the array. Then i ask for a return and used the .keys method which returns an array of the objects strings. i also used the .includes() method that looks for one string that could be in another string returning the true or false as it should. And the .toLowerCase converts a string to lower case letters
```

Comment:
It's clear that you write your pseudo code after the function, which is not what we teach. Also, based on your pseudo code, it also looks like you are taking code and just explaining it rather than coding out an approach on your own. If you don't understand the code, you are not learning or growing and only hurting yourself.
And also, the function as it is right now will produce an error, I suggest going back and figuring out what is going on so you can learn from it